userspace/libsinsp/mpsc_priority_queue.h

```diff
@@ -155,6 +155,22 @@ class mpsc_priority_queue
 		}
 	}
 
+	/**
+	 * @brief Sets the maximum capacity of the queue. Returns false
+	 * if the the specified capacity has not been set (when the current queue
+	 * size is bigger than the specified capacity). 
+	 */
+	inline bool set_capacity(size_t capacity)
+	{
+		if(m_queue.size() < capacity)
+		{
+			m_capacity = capacity;
+			return true;
+		}
```

Comment:
```suggestion
		std::scoped_lock<Mtx> lk(m_mtx);
		if(capacity >= m_capacity)
		{
			m_capacity = capacity;
			return true;
		}
```
From my understanding the goal is to not make the queue have a smaller capacity than its current one -- but since we cap it to the value of `m_capacity` why not using that for clarity? Also, setting the same capacity should return `true`. Locking is essential here, otherwise there's gonna be a data race with `push`.

Reply:
> From my understanding the goal is to not make the queue have a smaller capacity than its current one -- but since we cap it to the value of m_capacity why not using that for clarity?

My goal was to not make the queue have a smaller capacity than its current _size_.
In this was the queue capacity can be reduced, while still being bigger than the current queue size.

Reply:
Works for me! Still adopt the lock though.